docs/guides/src/main/server/enabletls.adoc

```diff
@@ -0,0 +1,70 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+
+<@tmpl.guide
+title="Configure Keycloak to use TLS"
+summary="Learn how to use HTTPS/TLS with Keycloak">
+
+Transport Layer Security (short: TLS) is crucial to exchange data over a secured channel. For production environments, you should never expose Keycloak endpoints through HTTP, as sensitive data is at the core of what Keycloak exchanges with other applications. In this guide you will learn how to configure Keycloak to use HTTPS/TLS.
+
+== Configure TLS in Keycloak
+Keycloak can be configured to load the needed certificate infrastructure either from a certificate / key file pair in PEM format or from an existing keystore file. When both alternatives are set up, the certificate / key file configuration takes precedence over the keystore configuration.
+
+=== Providing a certificate / key file pair
+When you use a pair of matching certificate / private key files in PEM format, configure Keycloak to use them by running:
+
+<@kc.all parameters="--https-certificate-file=/path/to/certfile.pem --https-certificate-key-file=/path/to/keyfile.pem"/>
+
+Keycloak creates a keystore out of these files in memory and uses this keystore afterwards.
+
+=== Providing a keystore
+As an alternative, you can use an existing keystore by running:
+<@kc.all parameters="--https-key-store-file=/path/to/existing-keystore-file"/>
+
+When no keystore file is configured explicitly, but http.enabled is set to false, Keycloak will try to use a `server.keystore` file in the `conf` subdirectory by default.
+
+[NOTE]
+No matter which alternative you choose for setting up the TLS configuration, please make sure to set a secure password for your keystore using the `https.key-store.password` option. We recommend to not set the password in plaintext via CLI or in `conf/keycloak.properties`, but instead using good practices such as using a vault or an environment variable - both can be substituted in the properties file. If no password is set, the default password `password` is used.
+
+== Enable other TLS Version(s)
+As per default, Keycloak uses the current TLS Version, which is 1.3 at the time of writing. If you need to use an older version, you may change the version by running:
+
+<@kc.all parameters="--https-protocols=<version>[,<version>]"/>
+
+To also allow TLSv1.2, you can use for example `kc.sh start --http-protocols=TLSv1.3,TLSv1.2`.
+
+== Switch the HTTPS port
+You can switch the port Keycloak listens on for HTTPS traffic by running:
+<@kc.all parameters="--https-port=<port>"/>
+
+== Using a truststore
```

Comment:
I'm not sure if this part is accurate.

The trust-store configuration here should be about setting up the HTTPS listener to verify client certificates.

The examples you are giving here are related to another level of configuration. Please, see https://www.keycloak.org/docs/latest/server_installation/#_truststore.

Reply:
also updated, pls check again.